src/ScVmm/ScVmm.Autorest/custom/New-AzScVmmVM.ps1

```diff
@@ -0,0 +1,677 @@
+    [System.Int32]
+    # Gets or sets the number of vCPUs for the vm.
+    ${CpuCount},
+
+    [Parameter(ParameterSetName='CreateExpandedByName')]
+    [Parameter(ParameterSetName='CreateExpandedById')]
+    [System.Management.Automation.SwitchParameter]
+    # Whether to enable dynamic memory or not.
+    ${DynamicMemoryEnabled},
+
+    [Parameter(ParameterSetName='CreateExpandedByName')]
+    [Parameter(ParameterSetName='CreateExpandedById')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.Int32]
+    # Gets or sets the max dynamic memory for the vm.
+    ${DynamicMemoryMaxMb},
+
+    [Parameter(ParameterSetName='CreateExpandedByName')]
+    [Parameter(ParameterSetName='CreateExpandedById')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.Int32]
+    # Gets or sets the min dynamic memory for the vm.
+    ${DynamicMemoryMinMb},
+
+    [Parameter(ParameterSetName='CreateExpandedByName')]
+    [Parameter(ParameterSetName='CreateExpandedById')]
+    [System.Management.Automation.SwitchParameter]
+    # Whether to enable processor compatibility mode for live migration of VMs.
+    ${LimitCpuForMigration},
+
+    [Parameter(ParameterSetName='CreateExpandedByName')]
+    [Parameter(ParameterSetName='CreateExpandedById')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.Int32]
+    # MemoryMB is the size of a virtual machine's memory, in MB.
+    ${MemoryMb},
+
+    [Parameter(ParameterSetName='CreateExpandedByName')]
+    [Parameter(ParameterSetName='CreateExpandedById')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [System.String]
+    # Type of checkpoint supported for the vm.
+    ${CheckpointType},
+
+    [Parameter(ParameterSetName='CreateExpandedByName')]
+    [Parameter(ParameterSetName='CreateExpandedById')]
+    [AllowEmptyCollection()]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Models.INetworkInterface[]]
+    # Gets or sets the list of network interfaces associated with the virtual machine.
+    ${NetworkInterface},
+
+    [Parameter(ParameterSetName='CreateExpandedByName')]
+    [Parameter(ParameterSetName='CreateExpandedById')]
+    [AllowEmptyCollection()]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Models.IVirtualDisk[]]
+    # Gets or sets the list of virtual disks associated with the virtual machine.
+    ${Disk},
+
+    [Parameter(ParameterSetName='CreateByName')]
+    [Parameter(ParameterSetName='CreateExpandedByName')]
+    [Parameter(ParameterSetName='CreateExpandedById')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Category('Body')]
+    [Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Runtime.Info(PossibleTypes=([Microsoft.Azure.PowerShell.Cmdlets.ScVmm.Models.IResourceUpdateTags]))]
+    [System.Collections.Hashtable]
+    # Resource tags
+    ${Tags},
```

Comment:
Parameter [-Tags] of cmdlet New-AzScVmmVM does not follow the enforced naming convention of using a singular noun for a parameter name. Please rename it to [-Tag]